### PR TITLE
Adds windows build

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

File: lib/mergeset/table.go [hunk 1]
```diff
@@ -1242,6 +1242,7 @@ func validatePath(pathPrefix, path string) (string, error) {
 	if err != nil {
 		return path, fmt.Errorf("cannot determine absolute path for %q: %w", path, err)
 	}
+
```
Comment (valyala) [inline lib/mergeset/table.go:1245]: Please remove this whitespace

File: lib/persistentqueue/persistentqueue.go [hunk 2]
```diff
@@ -213,7 +213,6 @@ func tryOpeningQueue(path, name string, chunkFileSize, maxBlockSize, maxPendingB
 		if !os.IsNotExist(err) {
 			logger.Errorf("cannot read metainfo for persistent queue from %q: %s; re-creating %q", metainfoPath, err, path)
 		}
-
```
Comment (valyala) [inline lib/persistentqueue/persistentqueue.go:216]: Please return back this whitespace

**Review by valyala — CHANGES_REQUESTED** [review]

File: lib/fs/fs_openbsd.go [hunk 3]
```diff
@@ -1,9 +1,65 @@
...
 
 import (
+	"fmt"
+	"os"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
 	"golang.org/x/sys/unix"
 )
 
+func mmap(fd int, offset int64, length int) (data []byte, err error) {
```
[the remark this answers is not in the document]
Comment (f41gh7) [reply lib/fs/fs_openbsd.go]: moved duplicate code into fs_unix.go with createFileLock.

File: lib/fs/fs.go [hunk 4]
```diff
@@ -12,24 +12,8 @@ import (
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/fasttime"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/filestream"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
-	"golang.org/x/sys/unix"
 )
 
-// MustSyncPath syncs contents of the given path.
-func MustSyncPath(path string) {
```
[the remark this answers is not in the document]
Comment (f41gh7) [reply lib/fs/fs.go:19]: leave public functions alone and create private OS-specific.

File: lib/procutil/signal_windows.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+func (sn *sigHUPNotifier) subscribe(sub chan<- os.Signal) {
+	sn.lock.Lock()
+	defer sn.lock.Unlock()
+	sn.subscribers = append(sn.subscribers, sub)
+}
+
+func (sn *sigHUPNotifier) notify(sig os.Signal) {
+	sn.lock.Lock()
+	sn.lock.Unlock()
```
Comment (valyala) [inline lib/procutil/signal_windows.go:53]:
```suggestion
	defer sn.lock.Unlock()
```

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
Thanks!

**Pull request merged**